cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -108,6 +111,19 @@ func (c *Controller) syncSpc(spc *apis.StoragePoolClaim) error {
 		glog.Errorf("Validation of spc failed:%s", err)
 		return nil
 	}
+	// spc finalizers should be removed only after deleting csp and removing
+	// finalizer on bdc which is added by spc
+	if isSPCDeletetionCandidate(spc) {
```

Comment:
nit: I liked @akhilerm approach in deciding this. Take a look if that makes sense to you.

Reply:
I will change to `spc.isSPCDeletetionCandidate()`